CRM/Civicpd/Page/CPDReport.php

```diff
@@ -1476,68 +1476,51 @@ function civi_cpd_report_update_activity()
 
 
     if (!empty($_POST['activity_id']) && !empty($_POST['category_id']) && civi_cpd_report_validate_date($_POST['credit_date']) && civi_cpd_report_validate_number($_POST['credits']) && !empty($_POST['notes']) && !empty($_POST['activity'])) {
-
-        $activityId = $_POST['activity_id'];
-
-        $startDate = (isset($_POST['start_date'])
-            && civi_cpd_report_validate_start_date($_POST['start_date']) ) ? "start_date='".$_POST['start_date']."'," : "start_date=NULL,";
-
-        $creditDate = $_POST['credit_date'];
-
-        $credits = number_format($_POST['credits'], 2, '.', '');
-
-        $activity = $_POST['activity'];
-
-        $notes = $_POST['notes'];
-
         $evidence = civi_cpd_report_import_activity_evidence_pdf();
-
-
         if ($evidence === false) {
-
             CRM_Civicpd_Page_CPDReport::redirectToReport();
-
         }
 
+        $updateCpdActivityQuery = "UPDATE civi_cpd_activities SET 
+                                   start_date = %1,
+                                   credit_date = %2,
+                                   credits = %3,
+                                   activity = %4,
+                                   notes = %5,
+                                   evidence = %6 
+                                   WHERE id = %7";
+
+        $startDate = NULL;
+        if (
+          !empty($_POST['start_date'])
+          && civi_cpd_report_validate_start_date($_POST['start_date'])
+        ) {
+            $startDate = $_POST['start_date'];
+        }
 
-        $sql = "
-
-            UPDATE civi_cpd_activities
-
-            SET
-
-                ".$startDate."
-
-                credit_date = '$creditDate',
-
-                credits = '$credits',
-
-                activity = '$activity',
-
-                notes = '$notes',
-
-                evidence = '$evidence'
-
-
-
-            WHERE id = $activityId";
-
+        $credits = number_format($_POST['credits'], 2, '.', '');
 
-        CRM_Core_DAO::executeQuery($sql);
+        $params = array(
```

Comment:
For this project to we need to use the `array()` syntax?

Reply:
yup, we usually use this syntax for public extensions and client projects (unless the project was decided to work on php >= 5.4 only such as civihr case)